### PR TITLE
Deprecate and archive camlp4 and web tutorials

### LINKED ISSUE #892: Web tutorial is somewhat out of date
See: https://ocaml.org/learn/tutorials/ocaml_and_the_web.html

I can't exactly remember the last time the idea of CGI programming entered my head, but I don't think it was this decade at least. So perhaps it would be a good idea to either remove it, or if there's a desire to preserve it as a historical curiosity, mark it as obsolete. It seems quite misleading and counterproductive to keep it as it is.

I would of course volunteer to write a new web tutorial if I thought I had a good enough grasp of the topic, but I came across this precisely because I don't.

### PR DESCRIPTION
**Pull request by glennsl**
Adds deprecation warnings to each tutorial page, and an "Archived Tutorials" section on the Tutorials index page.

Fixes #892

### CODE REVIEW

File: site/learn/tutorials/ocaml_and_the_web.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 *Table of contents*
 
 # OCaml and the Web
+
+**Deprecation Warning:** this tutorial is outdated and largely irrelevant unless you have an odd fascination with CGI Programming.
```
Comment (Chris00) [inline site/learn/tutorials/ocaml_and_the_web.md:7]: > unless you have an odd fascination with CGI Programming.

Maybe reword that part?  Also add that help is welcome?

**Comment by agarwal** [comment]
In the interest of moving on, I'll go ahead and merge this.
